src/main/java/org/gradle/profiler/AndroidStudioSyncAction.java

```diff
@@ -0,0 +1,56 @@
+package org.gradle.profiler;
+
+import com.android.builder.model.AndroidProject;
+import org.gradle.tooling.BuildController;
+import org.gradle.tooling.events.ProgressListener;
+import org.gradle.tooling.model.gradle.BasicGradleProject;
+import org.gradle.tooling.model.gradle.GradleBuild;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * A mock-up of Android studio sync.
+ */
+public class AndroidStudioSyncAction implements BuildAction {
```

Comment:
The way Android studio syncs is evolving quite rapidly, so what we do here is probably already outdated. I think I copied it when Android 3.0 came out. It would be great if we had a standalone library that we could just reference.

cc @ducrohet

Reply:
Yes we want to extract this code in a library. I think it'll happen in a few months.